tasks/sync-granule/GranuleFetcher.js

```diff
@@ -36,47 +29,61 @@ const addChecksumToFile = async (providerClient, checksumFiles, dataFile) => {
 };
 
 const addChecksumsToFiles = async (providerClient, files) => {
-  const checksumFiles = files.filter(isChecksumFile);
-  const dataFiles = files.filter((f) => !isChecksumFile(f));
-
-  const updatedDataFiles = await Promise.all(
-    dataFiles.map((dataFile) =>
-      addChecksumToFile(providerClient, checksumFiles, dataFile))
+  // Map data file name to checksum file object, if it has a checksum file
+  const checksumFileOf = files.reduce((acc, file) => {
```

Comment:
Consider `checksumFile` instead of `file` here. Which only really makes sense if you pull the filter out of the reducer, but I think that would be a good thing.